compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -539,45 +541,64 @@ def IREELinalgExt_AttentionOp : IREELinalgExt_Op<"attention",
     FlashAttention 2 and optionally results in the current `max` and `sum`
     statistics while processing the current tile.
 
-    If transpose_v is speciifed, the V tensor passed as input is assumed to
-    be transposed:
-
     attention(Q, K, V, scale) = softmax(Q @ K.T * scale) @ V.T
-
-    TODO: We should be moving to using a indexing map like approach so we
-    can generalize which tensor is transposed and which is not.
   }];
 
-  let arguments = (ins Variadic<AnyType>:$inputs,
+  let arguments = (ins AnyShaped:$query,
+                       AnyShaped:$key,
+                       AnyShaped:$value,
+                       AnyFloat:$scale,
                        Variadic<AnyShaped>:$outputs,
-                       DefaultValuedOptionalAttr<BoolAttr, "false">:$transpose_v
+                       AffineMapArrayAttr:$indexing_maps
   );
 
-  let builders = [
-    OpBuilder<(ins "ValueRange":$inputs, "ValueRange":$outputs)>,
-  ];
-
   let results = (outs Variadic<AnyRankedTensor>:$results);
   let hasFolder = 1;
+  let hasVerifier = 1;
+  let hasCustomAssemblyFormat = 1;
   let assemblyFormat = [{
     attr-dict
-    `ins` `(` $inputs `:` type($inputs) `)`
+    `ins` `(` $query `,` $key `,` $value `,` $scale `:` type($query) `,` type($key) `,` type($value) `,` type($scale) `)`
     `outs` `(` $outputs `:` type($outputs) `)`
     (`->` type($results)^)?
   }];
 
-  let extraClassDeclaration = extraLinalgExtOpClassDeclaration # [{
-    Value getQuery() {
-      return getDpsInputOperand(0)->get();
+  let extraClassDeclaration = [{
+    // Method to implement for specifying output range for
+    // DestinationStyleOpInterface
+    MutableOperandRange getDpsInitsMutable() {
+      return getOutputsMutable();
     }
-    Value getKey() {
-      return getDpsInputOperand(1)->get();
+
+    SmallVector<AffineMap> getIndexingMapsArray();
+
+    AffineMap getQueryMap() {
+      return cast<AffineMap>(getIndexingMapsArray()[0]);
+    }
+    AffineMap getKeyMap() {
+      return cast<AffineMap>(getIndexingMapsArray()[1]);
     }
-    Value getValue() {
-      return getDpsInputOperand(2)->get();
+    AffineMap getValueMap() {
+      return cast<AffineMap>(getIndexingMapsArray()[2]);
     }
-    Value getScale() {
-      return getDpsInputOperand(3)->get();
+    AffineMap getOutputMap() {
+      return cast<AffineMap>(getIndexingMapsArray()[3]);
+    }
+    int64_t getIterationDomainRank() {
+      return getQueryMap().getNumDims();
+    }
+
+    // TODO: Remove helper fns below once we move
```

Comment:
dumb question: is fns == functions? If so, can you replace it with `functions` which is more common in the codebase?

Reply:
Yeap! updated the comment, lmk if it's still too vague.